docs/api-reference/webgl/copy-and-blit.md

```diff
@@ -60,50 +58,33 @@ This function makes calls to the following WebGL APIs:
 
-Source options:
-  * `opts.source` (`Texture` or `Framebuffer`) - If provided this object will be bound and data copied from it.
-  * `opts.x` (`GLint`, optional, default: 0) - x coordinate of the lower left corner where to start copying.
-  * `opts.y` (`GLint`, optional, default: 0) - y coordinate of the lower left corner where to start copying.
+  * `source` (`Texture` or `Framebuffer`) - This object will be bound and data copied from it.
 
-Target options:
-  * `opts.texture` (`Texture`, optional) - If provided this object will be bound and data copied into it, if not provided `target` must be set and data will be copied to currenlty bound `Texture` object.
-  * `opts.target` (`GLenum`, optional) - Binding point where target texture is currently bound. Either `texture` or `target` must be specified.
-  * `opts.xoffset` (`GLint`, optional) - X offset with in target texture.
-  * `opts.yoffset` (`GLint`, optional) - Y offset with in target texture.
-  * `opts.zoffset` (`GLint`, optional, WebGL2) - Z offset with in target texture, when using copying into 2D Array of 3D texture.
-  * `opts.width` (`GLint`, optional, default: texture.width) - Width of the target texture.
-  * `opts.height` (`GLint`, optional, default: texture.height) - Height of the target texture.
+  Optional parameters:
+  * `opts.sourceAttachment` - (*GLenum*, default: `COLOR_ATTACHMENT0`) Used to deduce the `type` when not provided.
+  * `opts.targetMaxHeight` - (*number*, default: Number.MAX_SAFE_INTEGER) Maximum height of the image to be in returned Data URL.
 
-Notes:
-  * `xoffset`, `yoffset`, `zoffset` : when an offset is specified, it implies we are copying data into a sub region of the target texture and internally `gl.copyTexSubImage2D` or `gl.copyTexSubImage3D` are used based on the `target`, for these cases it is assumed that target texture has enough GPU memory already allocated. When none of the offsets are specified, `gl.copyTexImage2D` is used to copy data to entire target region and GPU memory is allocated if needed, target texture GPU memory doesn't have to be pre-allocated.
+Note:
+  * Works only under a browser environment, doesn't work under Node.
 
 This function makes calls to the following WebGL APIs:
 
-[`gl.copyTexImage2D`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/copyTexImage2D), [`gl.copyTexSubImage2D`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/copyTexSubImage2D) and [`gl,copyTexSubImage3D`](https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/copyTexSubImage3D)
+[`gl.readPixels`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/readPixels), [`gl.bindFramebuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindFramebuffer)
 
 
-### copyToDataUrl(opts: Object) : Data URL
+### readPixelsToImage(source : Framebuffer|Texture, opts: Object) : Image
```

Comment:
Again not sure readPixels is appropriate here. I guess it is possible to call `Image.getImageData` in the browser.

Does this function even work under Node.js?